composer_multi_project_monitoring/main.tf

```diff
@@ -0,0 +1,1018 @@
+#       2. Store the name of this project in "monitoring_project" variable below, in the locals block
+#       3. Save the list of projects with Cloud Composer environments to be monitored in "monitored_projects", set in the locals block
+#       4. Run "terraform apply"
+#
+#   The script creates the following resources in the monitoring project:
+#           1. Adds monitored projects to Cloud Monitoring
+#           2. Creates Alert Policies
+#           3. Creates Monitoring Dashboard
+#
+
+
+#######################################################
+#  
+# Overall settings
+#
+########################################################
+
+
+locals {
+    monitoring_project = "YOUR_MONITORING_PROJECT"
+    monitored_projects = toset(["YOUR_PROJECT_TO_MONITOR_1", "YOUR_PROJECT_TO_MONITOR_2", "YOUR_PROJECT_TO_MONITOR_3"])
+}
+
+#######################################################
+#  
+# Provider
+#
+########################################################
+
+provider "google-beta" {
+  project = local.monitoring_project
+}
+provider "google" {
+  project = local.monitoring_project
+}
+
+#######################################################
+#  
+# Add Monitored Projects to the Monitoring project
+#
+########################################################
+
+resource "google_monitoring_monitored_project" "projects_monitored" {
+  for_each = local.monitored_projects
+  metrics_scope = join("",["locations/global/metricsScopes/",local.monitoring_project])
+  name          = "${each.value}"
+  provider      = google-beta
+}
+
+
+#######################################################
+#  
+# Create alert policies in Monitoring project
+#
+########################################################
+
+resource "google_monitoring_alert_policy" "environment_health" {
+  display_name = "Environment Health"
+  combiner     = "OR"
+  conditions {
+    display_name = "Environment Health"
+    condition_monitoring_query_language {
+        query = join("", [
+            "fetch cloud_composer_environment",
+            "| {metric 'composer.googleapis.com/environment/dagbag_size'",
+            "| group_by 5m, [value_dagbag_size_mean: if(mean(value.dagbag_size) > 0, 1, 0)]",
+            "| align mean_aligner(5m)",
+            "| group_by [resource.project_id, resource.environment_name],    [value_dagbag_size_mean_aggregate: aggregate(value_dagbag_size_mean)];  ",
+            "metric 'composer.googleapis.com/environment/healthy'",
+            "| group_by 5m,    [value_sum_signals: aggregate(if(value.healthy,1,0))]",
+            "| align mean_aligner(5m)| absent_for 5m }",
+            "| outer_join 0",
+            "| group_by [resource.project_id, resource.environment_name]",
+            "| value val(2)",
+            "| align mean_aligner(5m)",
+            "| window(5m)",
+            "| condition val(0) < 0.9"
+            ])
+        duration = "120s"
+        trigger {
+            count = "1"
+        }
+    }
+  }
+
+
+  #alert_strategy {
```

Comment:
Any reason for this commented code (here and throughout)? Maybe adding a comment above will help.

Reply:
I think the original author included the strategy as an option for folks to uncomment if they wanted. I'll add a comment explaining that.